ecs-cli/modules/cli/attributechecker/attribute_checker_app.go

```diff
@@ -0,0 +1,171 @@
+	if err != nil {
+		logrus.Fatal("Error executing 'Attribute Checker': ", err)
+	}
+	commandConfig, err := config.NewCommandConfig(c, rdwr)
+	if err != nil {
+		logrus.Fatal("Error executing 'Attribute Checker': ", err)
+	}
+
+	ecsClient := ecsclient.NewECSClient(commandConfig)
+
+	taskDefAttributeNames, err := taskdefattributesCheckRequest(c, ecsClient)
+	if err != nil {
+		logrus.Fatal("Error executing 'Attribute Checker': ", err)
+	}
+	if len(taskDefAttributeNames) == 0 {
+		logrus.Info("The given task definition does not have any attributes")
+	}
+
+	descrContainerInstancesResponse, err := describeContainerInstancesAttributeMap(c, ecsClient, commandConfig)
+	if err != nil {
+		logrus.Fatal("Error executing 'Attribute Checker': ", err)
+	}
+
+	compareOutput := compare(taskDefAttributeNames, descrContainerInstancesResponse)
+	result := ConvertToInfoSet(compareOutput)
+	os.Stdout.WriteString(result.String(infoColumns, displayTitle))
+}
+
+func contains(containerInstanceAttributeNames []*string, tdAttrNames *string) bool {
+	for _, containerInstAttrNames := range containerInstanceAttributeNames {
+		if *containerInstAttrNames == *tdAttrNames {
+			return true
+		}
+	}
+	return false
+}
+
+//compares between container instances and Task definition
+func compare(taskDefAttributeNames []*string, descrContainerInstancesResponse map[string][]*string) map[string]string {
+	attributeCheckerResult := make(map[string]string)
+	for containerInstanceARN, containerInstanceAttributeNames := range descrContainerInstancesResponse {
+		var missingAttributes []string
+		for _, tdAttrNames := range taskDefAttributeNames {
+			if !contains(containerInstanceAttributeNames, tdAttrNames) {
+				missingAttributes = append(missingAttributes, *tdAttrNames)
+			}
+		}
+		missingAttributesNames := strings.Join(missingAttributes, ", ")
+		if len(missingAttributesNames) == 0 {
+			missingAttributesNames = "None"
+		}
+		containerInstance := strings.Split(containerInstanceARN, "/")
+		attributeCheckerResult[containerInstance[1]] = missingAttributesNames
+	}
+	return attributeCheckerResult
+}
+
+// DescribeContainerInstancesAttributeMap and get a map with Container instance ARN and Container instances attribute Names
+func describeContainerInstancesAttributeMap(context *cli.Context, ecsClient ecsclient.ECSClient, commandConfig *config.CommandConfig) (map[string][]*string, error) {
+	if err := validateCluster(commandConfig.Cluster, ecsClient); err != nil {
+		return nil, err
+	}
+	var containerInstanceIdentifiers []*string
+	containerInstanceIdentifier := context.String(flags.ContainerInstancesFlag)
+	splitValues := strings.Split(containerInstanceIdentifier, ",")
+	containerInstanceIdentifiers = aws.StringSlice(splitValues)
+
+	descrContainerInstancesAttributes, err := ecsClient.GetAttributesFromDescribeContainerInstances(containerInstanceIdentifiers)
+	if err != nil {
+		return nil, errors.Wrapf(err, fmt.Sprintf("Please check region/containerInstance values"))
```

Comment:
nit: I'd prefer "Failed to Describe Container Instances, please check region/containerInstance values"